src/Quadrature/quadrature.jl

```diff
@@ -169,15 +170,15 @@ the `:legendre` and `:lobatto` rules are implemented.
 
 `FacetQuadratureRule` is used as one of the components to create [`FacetValues`](@ref).
 """
-struct FacetQuadratureRule{shape, T, dim}
-    face_rules::Vector{QuadratureRule{shape, T, dim}}
-    function FacetQuadratureRule{shape, T, dim}(face_rules::Vector{QuadratureRule{shape, T, dim}}) where {shape, T, dim}
+struct FacetQuadratureRule{shape, T, dim, QRType <: QuadratureRule{shape, T, dim}, FacetRulesType <: Union{NTuple{<:Any, QRType}, AbstractVector{QRType}}}
+    face_rules::FacetRulesType
```

Comment:
For the `Array`-based `QuadratureRule` this only contains references to two arrays, so I think we can simplify to always use a homogeneous tuple. `nfaces(RefShape)` should give a Const return value if needed to get `N`. 
```suggestion
struct FacetQuadratureRule{RefShape, QR<:QuadratureRule{RefShape}, N}
    face_rules::NTuple{N, QR}
```

Reply:
If everyone is okay with this then we can also follow this idea. I mean even tesseract only has 8 facets.

Reply:
But does everything in Ferrite work if we use a non-indexable container? I mean, the array interface is the supertype for all indexable containers, right?